src/CMakeLists.txt

```diff
@@ -1565,6 +1565,10 @@ target_link_libraries(
 
 string(TIMESTAMP CP2K_TIMESTAMP "%Y-%m-%d %H:%M:%S")
 
+if(NOT CP2K_DATA_DIR)
```

Comment:
That's not a proper way to do it. If you want to change the default location for the data directory (and rightfully so) you have to add an compilation option to the CMakeLists.txt instead of relying on a variable that you define manually.

what i mean is this

`set(CP2K_DATA_DIR  "default" CACHE STRING "Set the location for cp2k data")` 

It has to moved to the main CMakeLists.txt (not the src/CMakeLists.txt)

then the if statement should be 

```
if (NOT CP2K_DATA_DIR MATCHES "default")
set(CP2K_DATA_DIR ${CMAKE_INSTALL_FULL_DATAROOTDIR}/cp2k/data)
endif
```
the rest is alright

Reply:
Need to drop the `NOT` here, right?

Reply:
Ah - option() is only for booleans.

Reply:
there is a simpler option than what I suggested. It is possible to set CMAKE_INSTALL_FULLDATAROOTDIR when you configure cp2k itself

`cmake -DCMAKE_INSTALL_FULLDATAROOTDIR=/usr/share ..` should give the expected behavior. I realized afterwards that this solution should be working. it means we do not need to set an option and we can keep ${CMAKE_INSTALL_FULL_DATAROOTDIR}/${PROJECT_NAME}/data

Reply:
${PROJECT_NAME} should  be defined by cmake automatically.

Reply:
> there is a simpler option than what I suggested. It is possible to set CMAKE_INSTALL_FULLDATAROOTDIR when you configure cp2k itself
> 
> `cmake -DCMAKE_INSTALL_FULLDATAROOTDIR=/usr/share ..` should give the expected behavior. I realized afterwards that this solution should be working. it means we do not need to set an option and we can keep `${CMAKE_INSTALL_FULL_DATAROOTDIR}/${PROJECT_NAME}/data` everywhere 

This isn't working for me:
```
+ /usr/bin/cmake -S . -B redhat-linux-build_mpich -DCMAKE_C_FLAGS_RELEASE:STRING=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE:STRING=-DNDEBUG -DCMAKE_Fortran_FLAGS_RELEASE:STRING=-DNDEBUG -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON -DCMAKE_INSTALL_DO_STRIP:BOOL=OFF -DCMAKE_INSTALL_PREFIX:PATH=/usr -DINCLUDE_INSTALL_DIR:PATH=/usr/include -DLIB_INSTALL_DIR:PATH=/usr/lib64 -DSYSCONF_INSTALL_DIR:PATH=/etc -DSHARE_INSTALL_PREFIX:PATH=/usr/share -DLIB_SUFFIX=64 -DBUILD_SHARED_LIBS:BOOL=ON -DCP2K_BLAS_VENDOR=FlexiBLAS -DCP2K_ENABLE_REGTESTS=ON -DCMAKE_PREFIX_PATH:PATH=/usr/lib64/mpich -DCMAKE_INSTALL_PREFIX:PATH=/usr/lib64/mpich -DCMAKE_INSTALL_FULL_DATAROOTDIR:PATH=/usr/share -DCMAKE_INSTALL_Fortran_MODULES:PATH=/usr/lib64/gfortran/modules/mpich/cp2k -DCMAKE_INSTALL_LIBDIR:PATH=lib -DCP2K_CMAKE_SUFFIX=_mpich -DCP2K_USE_MPI_F08:BOOL=ON
```
Still results in data being installed in `/usr/lib64/mpich/share/cp2k/data`.  This is because while GNUInstallDirs sets and uses this variable, it makes no provision for overriding it.